src/Traits/OptimusEncodedRouteKey.php

```diff
@@ -40,6 +40,31 @@ public function resolveRouteBinding($value, $field = null)
         return $this->where($field, $value)->first();
     }
 
+    /**
+     * Retrieve the model for a bound value.
+     *
+     * @param  \Illuminate\Database\Eloquent\Model|\Illuminate\Database\Eloquent\Relations\Relation  $query
+     * @param  mixed  $value
+     * @param  string|null  $field
+     * @return \Illuminate\Database\Eloquent\Relations\Relation
+     */
+    public function resolveRouteBindingQuery($query, $value, $field = null)
+    {
+        if ($field === null) {
+            $field = $query->getRouteKeyName();
+        }
+
+        if (is_string($value) && ctype_digit($value)) {
+            $value = (int) $value;
+        }
+
+        if (is_int($value) && $field === $this->getRouteKeyName()) {
+            $value = $this->getOptimus()->decode($value);
+        }
+
+        return $query->where($field ?? $this->getRouteKeyName(), $value);
```

Comment:
`$field` cannot be `null`, you are setting it on the line 54.

And if we remove null coalescence, then this method will be identical to the `resolveRouteBinding`.

Reply:
sorry fixed